addOns/reports/src/test/java/org/zaproxy/addon/reports/ExtensionReportsUnitTest.java

```diff
@@ -1019,6 +1006,127 @@ void shouldGenerateValidXmlReport() throws Exception {
+        assertThat(instanceChildNodes.item(y).getTextContent(), is(equalTo("Test <p>Evidence")));
+        y++;
+        assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("#text"))); // Filler
+        y++;
+        assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("otherinfo")));
+        assertThat(instanceChildNodes.item(y).getTextContent(), is(equalTo("Test 'Other\\")));
+        y++;
+        if (isXmlPlus) {
+            assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("#text"))); // Filler
+            y++;
+            assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("requestheader")));
+            assertThat(
+                    instanceChildNodes.item(y).getTextContent(),
+                    is(
+                            equalTo(
+                                    "GET http://example.com/example_3 HTTP/1.1\nHost: example.com\nUser-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:92.0) Gecko/20100101 Firefox/92.0\nPragma: no-cache\nCache-Control: no-cache\n\n")));
+            y++;
+            assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("#text"))); // Filler
+            y++;
+            assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("requestbody")));
+            assertThat(
+                    instanceChildNodes.item(y).getTextContent(), is(equalTo("Test Request Body")));
+            y++;
+            assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("#text"))); // Filler
+            y++;
+            assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("responseheader")));
+            assertThat(instanceChildNodes.item(y).getTextContent(), is(equalTo("HTTP/1.0 0\n\n")));
+            y++;
+            assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("#text"))); // Filler
+            y++;
+            assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("responsebody")));
+            assertThat(
+                    instanceChildNodes.item(y).getTextContent(), is(equalTo("Test Response Body")));
+            y++;
+        }
+        assertThat(instanceChildNodes.item(y).getNodeName(), is(equalTo("#text"))); // Filler
+    }
+
+    @Test
+    void shouldGenerateValidXmlReport() throws Exception {
+        // Given
+        Template template = getTemplateFromYamlFile("traditional-xml");
+        String fileName = "basic-traditional-xml";
+        File f = File.createTempFile(fileName, template.getExtension());
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+        DocumentBuilder db = dbf.newDocumentBuilder();
+
+        // When
+        File r = generateReportWithAlerts(template, f);
+        Document doc = db.parse(r);
+        Element root = doc.getDocumentElement();
+        checkXmlAlert(doc, false);
```

Comment:
Should be moved to `// Then`, same in following test.

Reply:
okk working on all